js/modules/tamagotchi.js

```diff
@@ -8,9 +8,12 @@ export default class Tamagotchi {
     this.hungerDecreaseInterval = setInterval(() => this.decreaseHunger(), 1000);
     this.energyDecreaseInterval = setInterval(() => this.decreaseEnergy(), 2000);
     this.funDecreaseInterval = setInterval(() => this.decreaseFun(), 1000);
+    this.stateCheckInterval = setInterval(() => this.stateCheck(), 1000)
```

Comment:
**issue:** You should not have to check your state on interval, because it can cause serious bugs if you'd like to parametrize your other interval values. Calling `this.stateCheck()` in constructor should do the job